test_scripts/Polices/External_UCS/External_Consent_OFF/002_ATF_Policies_External_Consent_OFF_disallowed_by_external_consent_entities_on_user_consent_disallowed.lua

```diff
@@ -0,0 +1,155 @@
+        externalConsentStatus = {}
+      }
+    })
+  :Do(function(_,data)
+      id_group_1 = common_functions_external_consent:GetGroupId(data, "ConsentGroup001")
+    end)
+end
+
+--------------------------------------------------------------------------
+-- Precondition:
+-- HMI sends OnAppPermissionConsent with consented function = disallowed
+--------------------------------------------------------------------------
+Test["TEST_NAME_OFF" .. "_Precondition_HMI_sends_OnAppPermissionConsent_userConsent"] = function(self)
+  local hmi_app_id_1 = common_functions:GetHmiAppId(config.application1.registerAppInterfaceParams.appName, self)
+  -- hmi side: sending SDL.OnAppPermissionConsent for applications
+  self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent", {
+      appID = hmi_app_id_1, source = "GUI",
+      consentedFunctions = {{name = "ConsentGroup001", id = id_group_1, allowed = false}}
+    })
+  self.mobileSession:ExpectNotification("OnPermissionsChange")
+  :ValidIf(function(_,data)
+      local validate_result = common_functions_external_consent:ValidateHMIPermissions(data,
+        "SubscribeWayPoints", {allowed = {}, userDisallowed = {"BACKGROUND","FULL","LIMITED"}})
+      return validate_result
+    end)
+end
+
+--------------------------------------------------------------------------
+-- Precondition:
+-- HMI sends OnAppPermissionConsent with External Consent status = OFF
+--------------------------------------------------------------------------
+Test["TEST_NAME_OFF" .. "_Precondition_HMI_sends_OnAppPermissionConsent_externalConsentStatus"] = function(self)
+  -- hmi side: sending SDL.OnAppPermissionConsent for applications
+  self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent", {
+      source = "GUI",
+      externalConsentStatus = {{entityType = 2, entityID = 5, status = "OFF"}}
+    })
+  self.mobileSession:ExpectNotification("OnPermissionsChange")
+  :ValidIf(function(_,data)
+      local validate_result = common_functions_external_consent:ValidateHMIPermissions(data,
+        "SubscribeWayPoints", {allowed = {"BACKGROUND","FULL","LIMITED"}, userDisallowed = {}})
+      return validate_result
+    end)
+end
+
+--------------------------------------------------------------------------
+-- Main check:
+-- RPC is allowed to process.
+--------------------------------------------------------------------------
+Test["TEST_NAME_OFF" .. "_MainCheck_RPC_is_allowed"] = function(self)
+  --mobile side: send SubscribeWayPoints request
+  self.mobileSession:SendRPC("SubscribeWayPoints",{})
+  --hmi side: expected SubscribeWayPoints request
+  EXPECT_HMICALL("Navigation.SubscribeWayPoints")
+  :Do(function(_,data)
+      --hmi side: sending Navigation.SubscribeWayPoints response
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS",{})
+    end)
+  --mobile side: SubscribeWayPoints response
+  EXPECT_RESPONSE("SubscribeWayPoints", {success = true , resultCode = "SUCCESS"})
```

Comment:
@istoimenova, Please update with usage of _correlationId_ because current version can be unstable